osu.Game.Modes.Osu/Objects/Slider.cs

```diff
@@ -25,17 +28,66 @@ public override int StackHeight
         }
 
         public double Velocity;
+        public double TickDistance;
 
         public override void SetDefaultsFromBeatmap(Beatmap beatmap)
         {
             base.SetDefaultsFromBeatmap(beatmap);
 
-            Velocity = 100 / beatmap.BeatLengthAt(StartTime, true) * beatmap.BeatmapInfo.BaseDifficulty.SliderMultiplier;
+            var baseDifficulty = beatmap.BeatmapInfo.BaseDifficulty;
+
+            var startBeatLength = beatmap.BeatLengthAt(StartTime);
+            var multipliedStartBeatLength = beatmap.BeatLengthAt(StartTime, true);
+
+            Velocity = 100 / multipliedStartBeatLength * baseDifficulty.SliderMultiplier;
+            TickDistance = (100 * baseDifficulty.SliderMultiplier) / baseDifficulty.SliderTickRate / (multipliedStartBeatLength / startBeatLength);

```

Comment:
Can we please not use / without parens? Also, this whole expression can easily be refactored to be less confusing. For instance by simply multiplying with `startBeatLength`. Lastly I recommend putting `/` to be the _last_ operator in the expression; otherwise someone who does not know operator ordering by heart will read something ambiguous.

Example:

2 / 2 * 4 = ?
(2 / 2) * 4 = 4
2 / (2 * 4) = 1/4